src/pages/index.js

```diff
@@ -123,7 +135,119 @@ function handleFormButton() {
   profileEditPopup.open();
 }
 
-function handleProfileSubmit({ title, description }) {
-  user.setUserInfo({ title, description });
+function handleProfileSubmit({ name, about }) {
+  profileEditPopup.saving(true);
+  api
+    .updateInfo(name, about)
+    .then((data) => {
+      user.setUserInfo(data);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => profileEditPopup.saving(false));
   profileEditPopup.close();
 }
+
+/* -------------------------------- Api Class ------------------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c6c362c5-eda0-4373-9004-84a264f04392",
+    "Content-Type": "application/json",
+  },
+});
+
+//Upload initial cards to the DOM
+api
+  .getInitialCards()
+  .then((cardData) => {
+    cardSection.renderItems(cardData);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+//Uploads user information from server to DOM
+api
+  .getUserInfo()
+  .then((userData) => {
+    user.setUserInfo(userData);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+//handler for like button
+function handleLikeClick(card) {
+  if (card._isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+  if (!card._isLiked) {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+}
+
+//select the delete confirmation modal
+const deleteConfirmModal = new PopupWithConfirmation({
+  popupSelector: "#card-delete-modal",
+});
+
+//handler for delete modal to appear
+function handleDeleteClick(card) {
+  deleteConfirmModal.open();
+  deleteConfirmModal.deleting(false);
+  deleteConfirmModal.handleDelete(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        // card.handleDelete(card._id);
+        deleteConfirmModal.close();
+        card._handleDeleteCard();
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(deleteConfirmModal.deleting(true));
+  });
+}
+
+//adds close with esc and close button
+deleteConfirmModal.setEventListeners();
+
+function handleAvatarSubmit(data) {
+  avatarPopup.saving(true);
+  api
+    .updateAvatar(data)
+    .then((res) => {
+      user.setUserAvatar(res.avatar);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarPopup.saving(false);
+      avatarPopup.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.